_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -1534,54 +1524,48 @@ mfxStatus VAAPIEncoder::CreateAuxilliaryDevice(
     m_caps.ddi_caps.ChromaWeightedPred      = 1;
     m_caps.ddi_caps.MaxNum_WeightedPredL0   = 4;
     m_caps.ddi_caps.MaxNum_WeightedPredL1   = 2;
-    m_caps.ddi_caps.Color420Only            = 1;
 
-    if ((attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != VA_ATTRIB_NOT_SUPPORTED) &&
-        (attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != 0))
-        m_caps.ddi_caps.MaxPicWidth  = attrs[idx_map[VAConfigAttribMaxPictureWidth]].value;
-    else
-        m_caps.ddi_caps.MaxPicWidth = 1920;
+    m_caps.ddi_caps.Color420Only = !(AV(VAConfigAttribRTFormat) & (VA_RT_FORMAT_YUV422 | VA_RT_FORMAT_YUV444));
 
-    if ((attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != VA_ATTRIB_NOT_SUPPORTED) &&
-        (attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != 0))
-        m_caps.ddi_caps.MaxPicHeight = attrs[idx_map[VAConfigAttribMaxPictureHeight]].value;
-    else
-        m_caps.ddi_caps.MaxPicHeight = 1088;
+    MFX_CHECK(AV(VAConfigAttribMaxPictureWidth) != VA_ATTRIB_NOT_SUPPORTED, MFX_ERR_UNSUPPORTED);
```

Comment:
It seems you change behavior here. Considering you introduce a fix in the same commit for the other staff, I strongly suggest to break this commit at least in 2, better 3.

Reply:
removed